src/icon.ts

```diff
@@ -47,7 +47,7 @@ export function icon(map: mapboxgl.Map, options?: Options) {
       imageLayout: map.getLayoutProperty('poi-level-1', 'icon-image'),
       textColor: map.getPaintProperty('poi-level-1', 'text-color'),
       ...options?.styles?.[StyleName.Icon],
-    },
+    } as Styles,
```

Comment:
I'd rather remove both `as Styles` and type the `const styles`:

```
const styles: { [key in StyleName]: Styles } = ...
```

Reply:
Done.